vochain/state.go

```diff
@@ -217,13 +219,23 @@ type State struct {
 }
 
 // NewState creates a new State
-func NewState(dataDir string) (*State, error) {
+func NewState(dataDir string, storageType db.StorageType) (*State, error) {
 	var err error
-	db, err := badgerdb.New(badgerdb.Options{Path: filepath.Join(dataDir, "vcstate")})
+	var kv db.Database
+	switch storageType {
+	case db.StoragePebble:
+		kv, err = pebbledb.New(pebbledb.Options{Path: filepath.Join(dataDir, "vcstate")})
+	case db.StorageBadgerv2:
+		kv, err = badgerdbv2.New(badgerdbv2.Options{Path: filepath.Join(dataDir, "vcstate")})
+	case db.StorageBadgerv3:
+		kv, err = badgerdbv3.New(badgerdbv3.Options{Path: filepath.Join(dataDir, "vcstate")})
+	default:
+		return nil, fmt.Errorf("storage type %d not recognized", storageType)
+	}
```

Comment:
if you put this "meta constructor" somewhere reusable, like in `db/metadb.New(storageType string)`, then the tests that currently hard-code pebble or badger could do something like:

```
metadb.New(os.Getenv("DB_BACKEND"))
```

`metadb.New` could also have a default when the string is empty, e.g. pebble.